src/System.Management.Automation/engine/ExecutionContext.cs

```diff
@@ -611,6 +602,11 @@ internal T GetEnumPreference<T>(VariablePath preferenceVariablePath, T defaultPr
                         result = (T)PSObject.Base(val);
                         defaultUsed = false;
                     }
+
+                    if (result is ActionPreference actionPreferenceValue)
+                    {
+                        CheckActionPreference(preferenceVariablePath, actionPreferenceValue, defaultPref);
+                    }
```

Comment:
Technically, this is the right fix. This makes it consistent for `$InformationPreference = "Ignore"; Write-Information abc` to run in global scope (ps console directly) and in local scope.
However, this will also be a breaking change for scripts that accidentally use `$InformationPreference = "Ignore"` in local scopes ...

Today it behaves inconsistently:
```powershell
PS:48> $InformationPreference = "Ignore"; Write-Information abc ## error out !
Write-Information : The value Ignore is not supported for an ActionPreference variable. The provided value should be used only as a value for a preference parameter, and has been replaced by the default value. For more information, see the Help topic, "about_Preference_Variables."
At line:1 char:36
+ $InformationPreference = "Ignore"; Write-Information abc
+                                    ~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Write-Information], NotSupportedException
+ FullyQualifiedErrorId : System.NotSupportedException,Microsoft.PowerShell.Commands.WriteInformationCommand

PS:49> & { $InformationPreference = "Ignore"; Write-Information abc }  ## no error ...
PS:50>
```

Reply:
As per my previous comments, this is a very temporary breaking change. PR #10238 removed the unnecessary restriction around `ActionPreference.Ignore`, as per [this discussion in issue 4348](https://github.com/PowerShell/PowerShell/issues/4348#issuecomment-490672315).

Getting this PR and #10238 reviewed and merged in eliminates the breaking change that resulted in me changing a Pester test in this PR.

Reply:
I think that fix should be incorporate into this PR. I looks to me we just need to remove the check of `preference == ActionPreference.Ignore` in `GetEnumPreference`.

Reply:
Done.